spectrum/input.py

```diff
@@ -202,13 +202,56 @@ def __init__(self, host, cdn_host):
 
     def session(self):
         browser = mechanicalsoup.Browser()
-        return JournalSession(self._host, browser)
+        return JournalHtmlSession(self._host, browser)
 
     def cdn_session(self):
         browser = mechanicalsoup.Browser()
-        return JournalSession(self._cdn_host, browser)
+        return JournalHtmlSession(self._cdn_host, browser)
 
-class JournalSession:
+    # TODO: rename
+    # TODO: allow CDN
+    def javascript_session(self, driver):
+        return JournalJavaScriptSession(driver, self._host)
+
+
+class JournalJavaScriptSession:
+    def __init__(self, driver, host):
+        self._driver = driver
+        self._host = host
+
+    def _log(self, message, *args, **kwargs):
+        LOGGER.info(message, extra={'app':'journal'}, *args, **kwargs)
+
+    def submit(self):
+        LOGGER.info("Loading: %s", self._host)
+        self._driver.get(self._host)
+        selenium_title_smoke_test('eLife', self._driver)
+
+        submit_link = self._driver.find_element_by_link_text('SUBMIT MY RESEARCH')
+        self._log("Found submit button")
+        submit_link.click()
+        self._log("Clicked submit button")
```

Comment:
```suggestion
        self._log("Clicked 'Submit my research' button")
```

(Any way to do a multi-line suggestion?)

Reply:
May work:
<img src="https://screenshotscdn.firefoxusercontent.com/images/43c06d93-4d91-4d26-8ab6-e1d2b0e17886.png" />
Will expand log with the new DOM id